test/e2e/azure_test.go

```diff
@@ -866,4 +866,56 @@ var _ = Describe("Workload cluster creation", func() {
 			By("PASSED!")
 		})
 	})
+
+	Context("Creating an aks cluster using clusterclass [Managed Kubernetes]", func() {
```

Comment:
@CecileRobertMichon In regards to writing an E2E test for this feature, would it be sufficient to just build a cluster and test using the existing AKS upgrade and machinepool tests like so? I think that testing the creation of a cluster using cluster class and then verifying the cluster came up should cover this feature, which the [existing cluster class test](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/test/e2e/azure_test.go#L823) does.